Update tutorial (#2225)
grafana/agent: docs/sources/flow/tutorials/collecting-prometheus-metrics.md

# Prerequisites

* [Docker](https://www.docker.com/products/docker-desktop)

# Scraping the Agent

> Run the following `curl https://raw.githubusercontent.com/grafana/agent/main/docs/sources/flow/tutorials/assets/runt.sh -O && bash ./runt.sh agent.flow`. 

Reviewer: Are users going to be comfortable running a script like this?  Just want to make sure in the long run that we're doing what people expect. I think what you had before was more typical, like in this tutorial: https://grafana.com/tutorials/play-with-grafana-mimir/.

Reviewer 2: Also, the current approach creates a bunch of files and folders in the current directory.

Personal preference, but how about adding a `mkdir flow_tutorials && cd flow_tutorials` first?

Author: I will add the `mkdir` and `cd.` Cloning and moving to the correct directory feels like a lot of friction to me. Will give it some thought to see if there is a cleaner method.

The `runt.sh` script does:

1. Downloads the configs necessary for Mimir, Grafana and the Grafana Agent. 
2. Downloads the docker image for Grafana Agent explicitly.
3. Runs the docker-compose up command to bring all the services up.

Allow the Grafana Agent to run for two minutes, then navigate to [Grafana](http://localhost:3000/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Mimir%22,%7B%22refId%22:%22A%22,%22instant%22:true,%22range%22:true,%22exemplar%22:true,%22expr%22:%22agent_build_info%7B%7D%22%7D%5D). 

![](../assets/agent_build_info.png)

This example scrapes the Grafana Agent's `http://localhost:12345/metrics` endpoint and pushes those metrics to the Mimir instance.

Navigate to `http://localhost:12345/graph` to view the Grafana Agent Flow UI. 

![](../assets/graph.png)

> 

Reviewer: ```suggestion
The Agent displays the component pipeline in a dependency graph.  See the [Scraping component](url) and [Remote Write ](url) sections for details about the components. 
```

Reviewer: Sorry, not sure yet about the hugo syntax...  I just want them to know that the details exist in the topic.

Reviewer: We should probably think about how best to arrange the topics in these tutorials -- a longer term goal. I wonder if we might want to create a pattern where we use expanding text near the top of a tutorial that would be titled "Understand the configuration" -- and we'd include the explanations of the components there.  We'd probably need to work with Jack, though?

Author: If we have expanding tags would love to use them, or spoiler style. I was going to put the answers to the tutorials in spoiler style sections but couldn't do it without custom CSS.

> Clicking on the nodes will navigate to the associated component page. This allows viewing the state, health information and if applicable the debug information.

Reviewer: ```suggestion
Click the nodes to navigate to the associated component page. There, you can view the state, health information, and, if applicable, the debug information.
```
